client/sound-system.cpp

```diff
@@ -93,6 +94,34 @@ namespace tec {
 	void SoundSystem::On(std::shared_ptr<EntityCreated> data) {
 		const proto::Entity& entity = data->entity;
 		eid entity_id = entity.id();
+		for (int i = 0; i < data->entity.components_size(); ++i) {
+			const proto::Component& comp = data->entity.components(i);
+			switch (comp.component_case()) {
+				case proto::Component::kAudioSource:
+				{
+					AudioSource* audio_source = new AudioSource();
+					audio_source->In(comp);
+					AudioSourceComponentMap::Set(entity_id, audio_source);
+				}
+				break;
+				case proto::Component::kRenderable:
+				case proto::Component::kPointLight:
+				case proto::Component::kDirectionalLight:
+				case proto::Component::kPosition:
+				case proto::Component::kOrientation:
+				case proto::Component::kView:
+				case proto::Component::kAnimation:
+				case proto::Component::kScale:
+				case proto::Component::kCollisionBody:
+				case proto::Component::kVelocity:
+				case proto::Component::kSpotLight:
+				case proto::Component::kVoxelVolume:
+				case proto::Component::kComputer:
+				case proto::Component::kLuaScript:
+				case proto::Component::COMPONENT_NOT_SET:
+				break;
```

Comment:
This kind of thing is a bit scary - is there a way we can avoid this?

Reply:
See below